cmake-tool/helpers/rootserver.cmake

```diff
@@ -101,37 +97,31 @@ function(DeclareRootserver rootservername)
             else()
                 set(march rv64imafdc)
             endif()
-            if(UseRiscVBBL)
-                # Package up our final elf image into the Berkeley boot loader.
-                # The host string is extracted from the cross compiler setting
-                # minus the trailing '-'
+            if(UseRiscVOpenSBI)
+                # Package up our final elf image into OpenSBI.
                 if("${CROSS_COMPILER_PREFIX}" STREQUAL "")
                     message(FATAL_ERROR "CROSS_COMPILER_PREFIX not set.")
                 endif()
 
-                string(
-                    REGEX
-                    REPLACE
-                        "^(.*)-$"
-                        "\\1"
-                        host
-                        "${CROSS_COMPILER_PREFIX}"
-                )
-                get_filename_component(host ${host} NAME)
+                if("${KernelOpenSBIPlatform}" STREQUAL "")
+                    message(FATAL_ERROR "KernelOpenSBIPlatform not set.")
+                endif()
+
                 file(GLOB_RECURSE deps)
+                set(
+                    OPENSBI_FW_PAYLOAD_ELF
+                    "${OPENSBI_PATH}/build/platform/${KernelOpenSBIPlatform}/firmware/fw_payload.elf"
+                )
                 add_custom_command(
-                    OUTPUT "${CMAKE_BINARY_DIR}/bbl/bbl"
-                    COMMAND mkdir -p ${CMAKE_BINARY_DIR}/bbl
+                    OUTPUT "${OPENSBI_FW_PAYLOAD_ELF}"
+                    COMMAND mkdir -p "${CMAKE_BINARY_DIR}/opensbi"
```

Comment:
Why are we still doing `mkdir -p "${CMAKE_BINARY_DIR}/opensbi"` here actually? Seem we can also drop this, because we do a `cd "${OPENSBI_PATH}` now to run the OpenSBI build